patches/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java.patch

```diff
@@ -0,0 +1,20 @@
+--- a/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
++++ b/net/minecraft/world/entity/ai/behavior/GiveGiftToHero.java
+@@ -121,8 +_,16 @@
+             return ImmutableList.of(new ItemStack(Items.POPPY));
+         } else {
+             VillagerProfession villagerprofession = p_23010_.getVillagerData().getProfession();
++            LootTable loottable = null;
```

Comment:
I just noticed that my earlier suggestion is backwards since the datamap should take precedence, that's my mistake 🤦. Please switch around the check to this instead:
```java
LootTable loottable = null;
var gift = net.minecraft.core.registries.BuiltInRegistries.VILLAGER_PROFESSION.wrapAsHolder(villagerprofession).getData(net.neoforged.neoforge.registries.datamaps.builtin.NeoForgeDataMaps.RAID_HERO_GIFTS);
if (gift != null) {
    loottable = p_23010_.level().getServer().getLootData().getLootTable(gift.lootTable());
} else if (GIFTS.containsKey(villagerprofession)) {
    loottable = p_23010_.level().getServer().getLootData().getLootTable(GIFTS.get(villagerprofession));
}
if (loottable != null) {
    ...
```